crud.py

```diff
@@ -0,0 +1,70 @@
+from sqlalchemy.orm import Session
+
+import models
+import schemas
+
+
+def get_all_authors(
+        db: Session,
+        skip: int | None,
+        limit: int | None = None,
+) -> list[models.Author]:
+    query = db.query(models.Author)
+
+    if limit:
+        query = query.offset(limit)
+
+    if skip:
+        query = query.offset(skip)
+
+    return query.all()
+
+
+def create_author(db: Session, author: schemas.AuthorCreate) -> models.Author:
+    db_author = models.Author(
+        name=author.name,
+        bio=author.bio
+    )
+    db.add(db_author)
+    db.commit()
+    db.refresh(db_author)
+
+    return db_author
+
+
+def get_author_detail(db: Session, id: int) -> models.Author:
+    return db.query(models.Author).filter(models.Author.id == id).first()
+
+
+def get_all_books(
+        db: Session,
+        skip: int | None,
+        limit: int | None,
+        author_id: int | None = None,
+) -> list[models.Book]:
+    query = db.query(models.Book)
+
+    if author_id:
+        query = query.filter(models.Book.author_id == author_id)
+
+    if limit:
+        query = query.offset(limit)
+
+    if skip:
+        query = query.offset(skip)
```

Comment:
And again, you are using the 'skip' variable to offset your query results in the 'get_all_books' function. This will offset your results twice if you had previously offset them. Please, check this logic.